src/packages/@ncigdc/modern_components/SampleType/SampleType.relay.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { graphql } from 'react-relay';
+import { compose, withPropsOnChange } from 'recompose';
+import Query from '@ncigdc/modern_components/Query';
+import { makeFilter } from '@ncigdc/utils/filters';
+import { withRouter } from 'react-router-dom';
+
+const fieldMap = {
+  sample: 'cases.samples.sample_id',
+  portion: 'cases.samples.portions.portion_id',
+  analyte: 'cases.samples.portions.analytes.analyte_id',
+  aliquot: 'cases.samples.portions.analytes.aliquots.aliquot_id',
+  slide: 'cases.samples.portions.slides.slide_id',
+};
+
+export default (Component: ReactClass<*>) =>
+  compose(
+    withRouter,
+    withPropsOnChange(['ae'], ({ ae }) => {
```

Comment:
now that it works, you can make this code self-documenting and more clear to future devs using this component if you only pass in what's needed

 ```Javascript
    withPropsOnChange(
      ['entityType', 'entityId'],
      ({ entityType, entityId }: { entityType: string, entityId: string }) => {
        return {
          variables: {
            filters: makeFilter([
              {
                field: fieldMap[entityType],
                value: entityId,
              },
            ]),
          },
        };
      },
    ),```